java/src/jmri/util/swing/JMenuUtil.java

```diff
@@ -192,5 +196,50 @@ static int convertStringToKeyEvent(@Nonnull String st) {
         return kcode;
     }
 
+    /**
+     * replace a menu item in its parent with another menu item
+     * <p>
+     * (at the same position in the parent menu)
+     *
+     * @param orginalMenuItem the original menu item to be replaced
+     * @param replacementMenuItem the menu item to replace it with
+     * @return true if the original menu item was found and removed or replaced
+     */
+    public static boolean replaceMenuItem(
+            @Nullable JMenuItem orginalMenuItem,
+            @Nonnull JMenuItem replacementMenuItem) {
+        return internalReplaceMenuItem(orginalMenuItem, replacementMenuItem);
+    }
+
+    /**
+     * remove a menu item from its parent
+     *
+     * @param theMenuItem the menu item to be replaced
+     * @return true if the menu item was found and removed
+     */
+    public boolean removeMenuItem(@Nullable JMenuItem theMenuItem)
+    {
+        return internalReplaceMenuItem(theMenuItem, null);
+    }
+
+    private static boolean internalReplaceMenuItem(
+            JMenuItem orginalMenuItem,
```

Comment:
You probably want to make the originalMenuItem non-null and place the responsibility of handling a null original menu item in making the decision to call this method, not in handling the output of this method.

Reply:
Ok, done!